zb-db/src/test/java/io/camunda/zeebe/db/impl/ColumnFamilyRandomizedPropertyTest.java

```diff
@@ -0,0 +1,228 @@
+              .isEqualTo(value);
+        });
+    columnFamily.forEach(
+        (key, value) -> assertThat(map).containsEntry(key.getValue(), value.getValue()));
+  }
+
+  record InsertOp(long key, long value) implements TestableOperation {
+
+    @Override
+    public BiConsumer<DbLong, DbLong> modify(final ColumnFamily<DbLong, DbLong> columnFamily) {
+      return columnFamily::insert;
+    }
+
+    @Override
+    public BiConsumer<Long, Long> modify(final Map<Long, Long> map) {
+      return map::putIfAbsent;
+    }
+  }
+
+  record PutOp(long key, long value) implements TestableOperation {
+
+    @Override
+    public BiConsumer<DbLong, DbLong> modify(final ColumnFamily<DbLong, DbLong> columnFamily) {
+      return columnFamily::put;
+    }
+
+    @Override
+    public BiConsumer<Long, Long> modify(final Map<Long, Long> map) {
+      return map::put;
+    }
+  }
+
+  record UpdateOp(long key, long value) implements TestableOperation {
+
+    @Override
+    public BiConsumer<DbLong, DbLong> modify(final ColumnFamily<DbLong, DbLong> columnFamily) {
+      return columnFamily::update;
+    }
+
+    @Override
+    public BiConsumer<Long, Long> modify(final Map<Long, Long> map) {
+      return (k, newValue) -> map.computeIfPresent(k, (k1, oldValue) -> newValue);
+    }
+  }
+
+  record UpsertOp(long key, long value) implements TestableOperation {
+
+    @Override
+    public BiConsumer<DbLong, DbLong> modify(final ColumnFamily<DbLong, DbLong> columnFamily) {
+      return columnFamily::upsert;
+    }
+
+    @Override
+    public BiConsumer<Long, Long> modify(final Map<Long, Long> map) {
+      return map::put;
+    }
+  }
+
+  record DeleteOp(long key, long value) implements TestableOperation {
+
+    @Override
+    public BiConsumer<DbLong, DbLong> modify(final ColumnFamily<DbLong, DbLong> columnFamily) {
+      return (k, v) -> columnFamily.delete(k);
+    }
+
+    @Override
+    public BiConsumer<Long, Long> modify(final Map<Long, Long> map) {
+      return (k, v) -> map.remove(k);
+    }
+  }
+
+  record DeleteIfExists(long key, long value) implements TestableOperation {
+
+    @Override
+    public BiConsumer<DbLong, DbLong> modify(final ColumnFamily<DbLong, DbLong> columnFamily) {
+      return (k, v) -> columnFamily.deleteIfExists(k);
+    }
+
+    @Override
+    public BiConsumer<Long, Long> modify(final Map<Long, Long> map) {
+      return (k, v) -> map.remove(k);
+    }
+  }
+
+  record DeleteExisting(long key, long value) implements TestableOperation {
+
+    @Override
+    public BiConsumer<DbLong, DbLong> modify(final ColumnFamily<DbLong, DbLong> columnFamily) {
+      return (k, v) -> columnFamily.deleteExisting(k);
+    }
+
+    @Override
+    public BiConsumer<Long, Long> modify(final Map<Long, Long> map) {
+      return (k, v) -> map.remove(k);
+    }
+  }
+
+  interface TestableOperation {
+    long key();
+
+    long value();
+
+    BiConsumer<DbLong, DbLong> modify(ColumnFamily<DbLong, DbLong> columnFamily);
```

Comment:
:wrench: Not sure why these are methods and not fields. As methods, since they are factory methods for a consumer, I find the name `modify` somewhat confusing. But this is more a nitpick